lib/Source.ml

```diff
@@ -367,3 +367,8 @@ let is_quoted_string t loc =
       | _ -> false)
   in
   not (List.is_empty toks)
+
+let loc_of_keyword t loc kwd =
+  match tokens_at t loc ~filter:(Poly.( = ) kwd) with
+  | [] -> None
+  | (_, loc) :: _ -> Some loc
```

Comment:
Just reading this function in isolation, it's not super clear whether this is intended to find one token or just return the first one. If it's just for one token, can you add an `assert false` in the non-singleton case; or if you're interested in the first one (I think it's that?), can you rename it to something like `loc_of_first_token_at`?

Reply:
done